File: CS/CustomXRLine/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Drawing.Design;

namespace CustomXRLine
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            XtraReport1 report = new XtraReport1();
            report.DesignerLoaded += new DevExpress.XtraReports.UserDesigner.DesignerLoadedEventHandler(report_DesignerLoaded);
            report.ShowDesigner();
        }

        void report_DesignerLoaded(object sender, DevExpress.XtraReports.UserDesigner.DesignerLoadedEventArgs e)
        {
            IToolboxService tb = e.DesignerHost.GetService(typeof(IToolboxService)) as IToolboxService;
            tb.AddToolboxItem(new ToolboxItem(typeof(CustomLine)));
        }
    }
}

File: VB/CustomXRLine/Form1.vb
Imports System
Imports System.Drawing.Design
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI

' ...
Namespace CustomXRLine

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim report As XtraReport1 = New XtraReport1()
            AddHandler report.DesignerLoaded, New DevExpress.XtraReports.UserDesigner.DesignerLoadedEventHandler(AddressOf report_DesignerLoaded)
            report.ShowDesigner()
        End Sub

        Private Sub report_DesignerLoaded(ByVal sender As Object, ByVal e As DevExpress.XtraReports.UserDesigner.DesignerLoadedEventArgs)
            Dim tb As IToolboxService = TryCast(e.DesignerHost.GetService(GetType(IToolboxService)), IToolboxService)
            tb.AddToolboxItem(New ToolboxItem(GetType(CustomLine)))
        End Sub
    End Class
End Namespace
